### DISCUSSION TITLE
How to use phoneme_language of 2 language

### DISCUSSION THREAD

**Discussion by harmlessman** [coqui-ai/TTS#1790] [General Q&A] [answered]
What I want is to apply phonemizer to each language.

TTS has a Japanese phonemizer by default.

I made a Korean phonemizer and actually used it, and it worked normally (only on the Korean dataset)

But the problem is when I learn multilingual.

If you put ja-jp (Japanese) in the phoneme_language of config, it doesn't become Korean phonemize, and if you put ko-kr, it's the same.

So, I tried to enter 'multi-language', which is the language of the multii-Phonemizer, but it didn't work out as I wanted.

What should I do to solve this?
(The Korean-phonemizer I made uses Hangeul as a phoneme. ex)"ㅏ", "ㄱ", "ㄴ")

And most of them are from open source content, but is it helpful for this TTS project to share the Korean-phonemizer that I made?

**Comment by erogol** [comment, marked as answer]
You need to use ```MultiPhonemizer``` however, I don't think we exposed it to the config level yet. You might need to do some coding to make it work. 

You need to define or edit new config fields that will be used in ```init_from_config``` of the model to initiate the right phonemizer with the right arguments. 

@Edresson do you know an easier way maybe?

**Reply by harmlessman** [reply]
Thank you for your answer.
The first time I've modified the `multiphonemizer` config `multiphonemizer phonemizer` in.
If you enter with korean japanese phonemizer to use in the same time.

The modifications include adding` MultiPhonemizer` from `phonemizer\__init__.py` to `get_phonemizer_by_name`.
and I made it possible to use `MultiPhonemizer` when `self.phonemizer.name()=='multi_phonemizer'` in `tokenizer.py text_to_ids` . 
Language is configured to distinguish between Korean and Japanese using `re`. 


But I don't think it's smart to configure it like this. So I'd like to ask you a question.
First of all, is there a way to get the language of the text I receive from `tokenizer.py text_to_id`?

Second, trying to configure and synthesize `MultiPhonemizer` directly results in an error that `MultiPhonemizer` does not exist. Looking at the error history, it seems to be an error caused by synthesis in `anaconda3\coqui-ts\synthesis.py`, not the project I am modifying, is there any way to do synthesis within the project?
For synthesis, type `python.\TTS\bin\synthesize.py --text "" --model_path ~`.

Finally, you said you need to edit init_from_config, but I don't have a clue.
In what direction should I edit?

Thank you for your hard work hard. Thank you.

**Reply by p0p4k** [reply]
Ah ha! I understand what you want @harmlessman  
- make a new conda environment
- git clone 🐸 TTS repo
- pip install -e .
- Now, if you pip list , you can see the installed directory for 🐸 TTS is the local folder where you cloned the repo.
- Make your changes in the local repo and they will reflect in the working.
Hope it helps!

**Reply by harmlessman** [reply]
Your answer is to the second question I asked. 
Similar to your answer, I ran `pip install -e .` in the directory and the synthesis using multi-phonemizer was successful!
Thank you for your reply.
